File: src/main/scala/exoskeleton/client/gui/GuiDebug.scala
package exoskeleton.client.gui

import java.util

import exoskeleton.common.lib.skills.PlayerSkills
import net.minecraft.client.gui.{GuiButton, GuiScreen}
import net.minecraft.entity.player.EntityPlayer

class GuiDebug(player: EntityPlayer)
extends GuiScreen{
  override def initGui(): Unit ={
    val list: util.List[GuiButton] = this.buttonList.asInstanceOf[util.List[GuiButton]];

    list.add(new GuiButton(1, 10, 35, 125, 20, "Add Night Vision"));
    list.add(new GuiButton(2, 10, 65, 125, 20, "Add No Drown"));
    list.add(new GuiButton(3, 10, 95, 125, 20, "Add Jump Boost"));
    list.add(new GuiButton(4, 10, 125, 125, 20, "Add Debug"));
    list.add(new GuiButton(9, 10, 155, 125, 20, "Add Breast Stroke"));

    list.add(new GuiButton(5, 137, 35, 125, 20, "Remove Night Vision"));
    list.add(new GuiButton(6, 137, 65, 125, 20, "Remove No Drown"));
    list.add(new GuiButton(7, 137, 95, 125, 20, "Remove Jump Boost"));
    list.add(new GuiButton(8, 137, 125, 125, 20, "Remove Debug"));
    list.add(new GuiButton(10, 137, 155, 125, 20, "Remove Breast Stroke"));
  }

  override def actionPerformed(button: GuiButton): Unit ={
    button.id match
    {
      case 1 =>{
        if(PlayerSkills.get(player).hasSkill("recon", "nightVision")){
        } else{
          PlayerSkills.get(player).addSkill("recon", "nightVision");
        }
      }
      case 2 =>{
        if(PlayerSkills.get(player).hasSkill("recon", "noDrown")){
        } else{
          PlayerSkills.get(player).addSkill("recon", "noDrown");
        }
      }
      case 3 =>{
        if(PlayerSkills.get(player).hasSkill("recon", "jumpBoost")){
        } else{
          PlayerSkills.get(player).addSkill("recon", "jumpBoost");
        }
      }
      case 4 =>{
        if(PlayerSkills.get(player).hasSkill("recon", "debug")){
        } else{
          PlayerSkills.get(player).addSkill("recon", "debug");
        }
      }
      case 5 =>{
        PlayerSkills.get(player).removeSkill("recon", "nightVision");
      }
      case 6 =>{
        PlayerSkills.get(player).removeSkill("recon", "noDrown");
      }
      case 7 =>{
        PlayerSkills.get(player).removeSkill("recon", "jumpBoost");
      }
      case 8 =>{
        PlayerSkills.get(player).removeSkill("recon", "debug");
      }
      case 9 =>{
        if(!PlayerSkills.get(player).hasSkill("recon", "breastStroke")){
          PlayerSkills.get(player).addSkill("recon", "breastStroke");
        }
      }
      case 10 =>{
        PlayerSkills.get(player).removeSkill("recon", "breastStroke");
      }
      case _ =>{

      }
    }
  }
}

File: src/main/scala/exoskeleton/client/gui/GuiDebug.java
package exoskeleton.client.gui;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import exoskeleton.api.Core;
import exoskeleton.api.ExoskeletonCores;
import exoskeleton.common.core.CoreAssassin$;
import exoskeleton.common.core.CoreBulldozer$;
import exoskeleton.common.core.CoreGhost$;
import exoskeleton.common.core.CoreInferno$;
import exoskeleton.common.core.CoreMystic$;
import exoskeleton.common.core.CoreRecon$;
import exoskeleton.common.core.CoreReflex$;
import exoskeleton.common.core.CoreSkybound$;
import net.minecraft.client.gui.GuiButton;
import net.minecraft.client.gui.GuiScreen;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;

@SideOnly(Side.CLIENT)
public class GuiDebug
extends GuiScreen{
    private final EntityPlayer player;

    public GuiDebug(EntityPlayer player){
        this.player = player;
    }

    public void initGui(){
        this.buttonList.add(new GuiButton(0, 10, 10, "Recon"));
        this.buttonList.add(new GuiButton(1, 10, 30, "Reflex"));
        this.buttonList.add(new GuiButton(2, 10, 50, "Inferno"));
        this.buttonList.add(new GuiButton(3, 10, 70, "Ghost"));
        this.buttonList.add(new GuiButton(4, 10, 90, "Skybound"));
        this.buttonList.add(new GuiButton(5, 10, 110, "Bulldozer"));
        this.buttonList.add(new GuiButton(6, 10, 130, "Mystic"));
        this.buttonList.add(new GuiButton(7, 10, 150, "Assassin"));
    }

    @Override
    public void actionPerformed(GuiButton button){
        ItemStack stack = this.player.getCurrentEquippedItem();

        if(stack == null){
            return;
        }

        Core core = ExoskeletonCores.findCore(stack);

        if(core != null){
            return;
        }

        switch(button.id)
        {
            case 0:{
                ExoskeletonCores.installCore(stack, CoreRecon$.MODULE$);
                break;
            }

            case 1:{
                ExoskeletonCores.installCore(stack, CoreReflex$.MODULE$);
                break;
            }

            case 2:{
                ExoskeletonCores.installCore(stack, CoreInferno$.MODULE$);
                break;
            }

            case 3:{
                ExoskeletonCores.installCore(stack, CoreGhost$.MODULE$);
                break;
            }

            case 4:{
                ExoskeletonCores.installCore(stack, CoreSkybound$.MODULE$);
                break;
            }

            case 5:{
                ExoskeletonCores.installCore(stack, CoreBulldozer$.MODULE$);
                break;
            }

            case 6:{
                ExoskeletonCores.installCore(stack, CoreMystic$.MODULE$);
                break;
            }

            case 7:{
                ExoskeletonCores.installCore(stack, CoreAssassin$.MODULE$);
                break;
            }
        }

        this.mc.setIngameFocus();
    }
}